### ISSUE TITLE
src/capnp/rpc-prelude.h:33:30: fatal error: persistent.capnp.h: No such file or directory

### ISSUE THREAD

**Issue by glycerine** [sandstorm-io/capnproto#216]
The latest cb7f2d048a14a70f84f8d6c91891c2bbc16d8a6b capnproto source seemed to compile fine on my osx box that had an earlier version of capnpc compiler available, but a fresh install on a new linux box seems to error out from a missing persistent.capnp.h file. Is there a bootstrap requirement now to have capnpc from a previous version installed?  Seems like that is less desirable and should be avoidable.  When I copied over the osx persistent.capnp.h and persistent.capnp.c++ to the linux box (they were never generated on the linux build), then the linux 'make' build finishes fine. 

~~~
$ make
...
libtool: compile:  g++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I./src -I./src -DKJ_HEADER_WARNINGS -DCAPNP_HEADER_WARNINGS -DCAPNP_INCLUDE_DIR=\"/usr/local/include\" -pthread -O2 -DNDEBUG -pthread -MT src/capnp/dynamic-capability.lo -MD -MP -MF src/capnp/.deps/dynamic-capability.Tpo -c src/capnp/dynamic-capability.c++ -o src/capnp/dynamic-capability.o >/dev/null 2>&1
depbase=`echo src/capnp/rpc.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;    /bin/sh ./libtool --tag=CXX   --mode=compile g++ -std=gnu++11 -DHAVE_CONFIG_H -I.    -I./src -I./src -DKJ_HEADER_WARNINGS -DCAPNP_HEADER_WARNINGS -DCAPNP_INCLUDE_DIR='"/usr/local/include"' -pthread -O2 -DNDEBUG -pthread -MT src/capnp/rpc.lo -MD -MP -MF $depbase.Tpo -c -o src/capnp/rpc.lo src/capnp/rpc.c++ &&    mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I./src -I./src -DKJ_HEADER_WARNINGS -DCAPNP_HEADER_WARNINGS -DCAPNP_INCLUDE_DIR=\"/usr/local/include\" -pthread -O2 -DNDEBUG -pthread -MT src/capnp/rpc.lo -MD -MP -MF src/capnp/.deps/rpc.Tpo -c src/capnp/rpc.c++  -fPIC -DPIC -o src/capnp/.libs/rpc.o
In file included from src/capnp/rpc.h:30:0,
                 from src/capnp/rpc.c++:22:
src/capnp/rpc-prelude.h:33:30: fatal error: persistent.capnp.h: No such file or directory
 #include "persistent.capnp.h"
                              ^
compilation terminated.
make[1]: *** [src/capnp/rpc.lo] Error 1
make[1]: Leaving directory `/fs1/raida/home/jason/go/src/github.com/glycerine/goq/vendor/capnproto/c++'
make: *** [all] Error 2
~/go/src/github.com/glycerine/goq/vendor/capnproto/c++ (master) $
~~~

**Comment by kentonv** [comment]
`persistent.capnp.{h,c++}`, like all other bootstraps, is checked into the repo:

https://github.com/sandstorm-io/capnproto/blob/master/c++/src/capnp/persistent.capnp.h

So it sounds like you somehow had an inconsistent or incomplete checkout on Linux. I'm not sure how that would happen, though...

(Aside: Old versions of the code generator actually won't work with newer versions of the library. We definitely don't require that one be installed; in fact we recommend that it not be.)

**Comment by glycerine** [comment]
I had done a git pull on a repo that was from a month or two ago. Maybe some kind of interaction with the .gitignore on pull?  Meh. Anyway, works from completely fresh sources, so closing this issue. Thanks Kenton.

**Comment by kentonv** [comment]
@glycerine It looks like `.gitignore` contained patterns that would match the bootstraps. I don't know if that would have caused your issue but it's definitely wrong. Fixed in commit 1d0b60c. Thanks.
